### PR TITLE
New map: Line Station

### PR DESCRIPTION
**Pull request by adacovsk**
[general]
- Redesign of old maps featuring long hallway, loaded with familiar vaults, departments modified to fit more snugly together. A Frankenstein monster. New to github, please be gentle.

[link to z1 of map](https://mdb.affectedarc07.co.uk/Files/10441188/4306460246/0/after.png)
[updated Nov 25 2021 z1](https://imgur.com/WoB0CLx.jpg)
[updated Nov 30 2021](https://i.imgur.com/FrKfnaB.jpg)

:cl:
rscadd: A new class of station has been commissioned from the Nanotrasen salvage yard and is ready for trials. Welcome to Frankenline Station.

### CODE REVIEW

**Comment by SonixApache** [comment]
i for one absolutely hate it, good job

**Comment by adacovsk** [comment]
Your feedback is appreciated. If you're not going to give some valid criticism, I really can't work on improving it.

**Comment by Zth--** [comment]
Add the render to OP, here's the [link](https://mdb.affectedarc07.co.uk/Files/10441188/4306460246/0/after.png)

**Comment by SootyBear42** [comment]
Actual frankenstein, morbidly curious.

**Comment by WindowsErrors** [comment]
if it's a compilation of all our favorite maps then I suppose it can't be any worse than island or horizon

**Comment by WindowsErrors** [comment]
oh i got an idea, you should add those conveyer belt walking things that they have at the airport to speed things along

**Comment by Eneocho** [comment]
You actually done fucking did it. Based and impossibly redpilled.
Now for the things:
The windows and airlocks separating both sides of the main hallway feel superfluous (and the scrubbers inside those windows do as well). The way it is currently is just begging for a mass shooting, since you can't yakkety sax your way around booletz/lasers with only a 2wide hall.
Medical is literally xoq's medical.
Library is box's library (I like the solitary reading rooms though).
Tech storage is box's.
Mechanics is xoq's.
Bridge is metaclub's.
I like cargo's pointlessly large backroom.
The scrubbers inside windows are weird.
Botany ranch room is way too fucking big for any practical use.
Could you move the xenobotany machines to main botany room?
Engineering is kinda maze-like and I don't like that.
AME room is too large (unless you also mapped in extra AME pieces/tanks).
There's almost no maintloot.
The Kitchen is waaay too far away from regularly transited areas. You have to go out of your way to get food and that disrupts flow.
Other things but my eyes hurt.

Can't be worse than Island/10.

**Comment by adacovsk** [comment]
> oh i got an idea, you should add those conveyer belt walking things that they have at the airport to speed things along

While I appreciate function over form, that will look awful. One feature is that from arrivals, you can go to cargo by trash dunking yourself to get to the starboard side, then go to disposals all the way back near arrivals, it is a secret escape mechanism, clowns rejoice, as it's right beside your office

>The scrubbers inside windows are weird.

Good eye, didn't see them

Also, I could just offset the entire station by 2 vertically from the middle to make the hallways wider... I removed the airlocks for more flow between the adjacent halls. Regarding the kitchen, I think it's more a feature than a design flaw, this was my intention. The great thing about this map is I can just move things around relatively easily, so the kitchen can be moved. Regarding engineering, it becomes more clear once you see it as atmos and engineering split on either side. Perhaps there's a better way to fit it all together by completely redoing atmos, but that's not something I wanted to do right now. I'm hesitant to make these changes at the moment but definitely can implement in the future. I did everything else you mentioned. Thanks!

Now how do I replace it on Github?

**Comment by Eneocho** [comment]
>Now how do I replace it on Github?

Take the new file and upload just like you did when making the PR, it should replace the older file.

**Comment by CptWad** [comment]
I cant wait for all the people complaining about getting lost
bagel was a circle and its incredible how clueless people can be when it isn't box

**Comment by adacovsk** [comment]
Did I do it right?

**Comment by adacovsk** [comment]
>I cant wait for all the people complaining about getting lost

It's literally a line. I refuse to believe this

**Comment by ghost** [comment]
this is incredibly cursed, bravo

**Comment by DeityLink** [comment]
of all the virology labs you picked Box's 😔 (Meta, Roid, and Lowfat Bagel viros are vastly improved.)

**Comment by agentofzog2** [comment]
Add ricochet rifles to the vault.
Also add another station's maint. Ideally deff's..

**Comment by Eneocho** [comment]
> of all the virology labs you picked Box's 😔 (Meta, Roid, and Lowfat Bagel viros are vastly improved.)

It's supposed to be a cursed mishmash. It fits the theme.

**Comment by DeityLink** [comment]
It can be cursed but still enjoyable to play

**Comment by ShiftyRail** [comment]
This is an absolutely awful idea. The map has no originality on its own other than the long hallway which is an inefficient and frustrating way to go for point A to point B - especially if someone has the funny idea of walling it, which they WILL, or if sec goes gestapo and install checkpoints with barriers, tape.

It also appears to be highly vulnerable to ZAS. Some détours and alternate paths via maint exist, but if the main hallway is vented (which it will), going from a South part of the station to a North part will be extremely painful and might be impossible for most jobs.

Responding effectively to emergencies - as sec, engineer, medical, or anything - is going to be very challenging and probably not worth the effort after a few tries. It's one thing to move away from Box's compact and easily patrollable format, but this one is too far.

I don't think it's worth adding to maps and to the compilation time of the server. We have no shortage of meme maps already.

**Comment by adacovsk** [comment]
> of all the virology labs you picked Box's 😔 (Meta, Roid, and Lowfat Bagel viros are vastly improved.)

I've played plenty of virology and never really noticed a difference, but will do that.

>Add ricochet rifles to the vault.
 
Will do. Already added lots of deff maint... Can try to fit in more.

> long hallway which is an inefficient and frustrating way to go for point A to point B - especially if someone has the funny idea of walling it, which they WILL, or if sec goes gestapo and install checkpoints with barriers, tape.

Funnily enough, the station is about the length of box. Disposals can be used for travel, I've added in a few to promote this. You can maint crawl from either fore or aft to get to any place (similar to box). There is lots of redundancy in the pipes and wiring in the main hall to prevent critical failure of the entire station from a bomb. If the glass windows become a problem, they can be changed in the future. Maintenance tunnels are next to every department.

Catbeasts have the most difficulty starting out of any antag, and I kept this in mind when designing. They can be anywhere on the map relatively quickly without being instantly killed now.

**Comment by ShiftyRail** [comment]
Most of Box's flow is within the central square. Which is about a third of the overall width.

**Comment by adacovsk** [comment]
> Most of Box's flow is within the central square. Which is about a third of the overall width.

That's true for cargo, AI, EVA, and command. It's not true for any other department. Line's hallway area is 202x5 = 1010, Box's "central hallway" is 25x3x2+ 34x3x2 = 354. Including the other hallways: sec 34x3, eng, 25x3, medbay to sci 56x3, lockeroom and tool 35x3. Grand total of 804. Box's arrivals makes it extend much further, but I'm not going to include it.  It depends on the metric but width isn't a fair comparison. If you ignore the windows in the middle of hallway you can get it down to 808.

**Comment by ghost** [comment]
uhhh youre PRing the same map twice, one in the main directory and another in maps/, pls remove the first one.

**Comment by Rabbidx7** [comment]
Love it. Make the bar not shit by switching the location of the counter. Either make it where the kitchen is now, and connect hydro to the kitchen through the livestock area, or make the bar at the top. If you go with the first suggestion though, it will allow people to grab food from the hall and not be forced to go into the bar, its a good change, imo.

**Comment by PrimeDSS13** [comment]
I personally believe it would be better as a bus map than a votable map due to the very nature of it, but I am interested on how it would play out.

**Comment by calzilla1** [comment]
The year is 2032: Vgstation now has 281 maps with increasingly elaborate gimmicks. Box, Deff, and Roid are still played half the time anyways

**Comment by adacovsk** [comment]
> I personally believe it would be better as a bus map than a votable map due to the very nature of it, but I am interested on how it would play out.

I think I settled on the name, "Frankenline Station" and could be played around Halloween. I could remove some lights and add more mobs to make it more spooky. If people like it enough, either way is nice.

**Comment by SonixApache** [comment]
i don't know if anyone's told you this yet but 
when self-hosting, use the Debug-Verbs verb
then under the [mapping] tab, there are a bunch of verbs that will help you find issues with your map
try 'em out

**Comment by adacovsk** [comment]
>i don't know if anyone's told you this yet but

Thanks. Just familiarized myself with the admin commands  and hosting earlier today. Close to resolving issues. trader and sci huttles not moving. Need more lights. Close to playable (imo). Maybe another day or two.

**Review by ShiftyRail — CHANGES_REQUESTED** [review]
The 'next_map' datum which controlls if this map can be voted or not is missing.

**Comment by ShiftyRail** [comment]
Probably need a 15 to 20 minimum players. The decision whether it should be enabled or not is a political one. I think it's a map best suited for buses if we want to merge it.
Also, you can add it to the CI flow by checking #31323 and making a similar comment for your map.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ env:
   BYOND_MINOR: 1566
 
   #This can be declared here because build.py will ignore it if DM_UNIT_TESTS is true.
-  ALL_MAPS: tgstation metaclub defficiency packedstation roidstation nrvhorizon test_box test_tiny snaxi tgstation-sec tgstation-snow LampreyStation xoq boxesstation synergy bagelstation lowfatbagel Dorfstation waystation island
```
Comment (ShiftyRail) [inline .github/workflows/ci.yml:27]: This apparently removed Island from CI, it didn't add linestation.

Comment (adacovsk) [reply .github/workflows/ci.yml:27]: A mix up at the cookie factory, it should be resolved now

**Review by ShiftyRail — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Eneocho) [reply .github/workflows/ci.yml:27]: > This apparently removed Island from CI

The hero we didn't deserve but the one we needed.

**Comment by Viciousaurus** [comment]
Can I fast travel from one end to the other when enemies are nearby?

**Comment by adacovsk** [comment]
> Can I fast travel from one end to the other when enemies are nearby?

if you're quick enough

**Comment by Eneocho** [comment]
Add a mass driver and pod console for shits and giggles so you can mass drive yourself from one side of the station to the other.

**Comment by ghost** [comment]
hallways and crossing points deserve directional arrows so hall monitors can tell people off for walking around the station in the wrong direction.

**Comment by adacovsk** [comment]
what decal would I use? The reason I ask is because most arrow decals look terrible and I don't want to use them

**Comment by Eneocho** [comment]
Just use white crayon arrows for that extra gunk feel.
Alternatively, the arrow warning stripe decals or the lame generic arrow decal.

**Comment by KillFurries** [comment]
It's not a bad map idea but needs more support. The entire map centers around the hallway and there being a long way to go from one side of the station to the other. The hallway is where most of the action will take place and it's where people will put their little autism project. This hallway is barren and it's the same empty hallway copypasted the entire way. No walls to break up line of sight, no vending machines to block shots from the other side of the station and to grab a drink while walking down the hallway. The different air locks are placed in awkward places that don't really make sense and it doesn't look like any thought has been put into the spacing and where the foot traffic will be.

**Comment by adacovsk** [comment]
> No walls to break up line of sight, no vending machines to block shots from the other side of the station and to grab a drink while walking down the hallway. The different air locks are placed in awkward places that don't really make sense and it doesn't look like any thought has been put into the spacing and where the foot traffic will be.

I think you're missing the point, but I am receptive of feedback regarding where the airlocks/foot traffic will be. There's actually a limited number of spots where the airlocks can go without looking stupid. We'll see people's thoughts on the placement when it's played and that's an easy fix.

**Comment by WindowsErrors** [comment]
the solution to this is simple: just add a second or a third long line hallway on the top or bottom of the station

**Comment by Eneocho** [comment]
Fuck it, hallway station. No access requirements.

**Comment by adacovsk** [comment]
> Fuck it, hallway station. No access requirements.

you've gone mad

**Comment by Eneocho** [comment]
> > Fuck it, hallway station. No access requirements.
> 
> you've gone mad

We've had hallway bars (and they tend to be a success), lets just have a hallway medbay, a hallway engineering and a hallway cargo.

**Comment by Eneocho** [comment]
Prepare for UNFORESEEN CONSEQUENCES

**Comment by WindowsErrors** [comment]
there was a round of line today

the mechanics hacked open all the airlocks down the hall and set up a giant mass driver for fast travel, which we subsequently loaded with a bunch of kitchen knives, toolboxes, fire extinguishers, and other sharp objects so that anyone who used them would get hurt, the medbay was filled with so many bodies

you know how security always seems to fire their weapons down the hall when they're chasing someone and they always friendly fire a crew member because of it? please stop doing that, you have a stun baton, we even got a couple clowns doing it intentionally with a ricochet rifle from the vault safe

we voted for line at the end of the round anyways

this is probably the best possible outcome for your map, it's totally better than island and horizon and taxi all put together

**Pull request merged**